### ISSUE TITLE
Identified in text AND described in text? Nr. 3: Problem with 2 statements in 1 SC: 3.3.1: Error Identification

### ISSUE THREAD

**Issue by jake-abma** [w3c/wcag#1810]
Back on this problem with the following:

Is what @JAWS-test wrote correct? and are 3.3.1 both statements for **TEXT**?

https://github.com/w3c/wcag/issues/1774#issuecomment-830508435 

> If an input error is automatically detected,
> 
> - the item that is in error is identified to the user in text
> - and the error is described to the user in text

If so I have two examples which boggle the mind!

Example 1 from Deque Demo pages: https://broken--deque-workshop.netlify.app/ 

- Click on Edit ingredients
- Leave an input empty
- click on SAVE
- See error message

<img width="474" alt="Screen Shot 2021-05-17 at 10 54 42 AM" src="https://user-images.githubusercontent.com/6384002/118461800-55101d80-b6fe-11eb-99cd-65b408bcf771.png">

**QUESTION:** This IS "the error is described to the user in text" BUT is "the item that is in error is identified to the user in text"?

Required was there already, the text doesn't say 'error' or alike, red is not 'text', and the "aria-invalid="true" doesn't count... so is it a FAIL?

Only 50% of the SC is done properly?

Example 2 from Deque Demo pages: https://broken--deque-workshop.netlify.app/ 

- Click on "Cooking Chocolate Cake"
- Fill in a negative number for "Rate the yumminess (0 - 50)"
- Click on "I COOKED IT"
- See the "error" message

**QUESTION:** This IS "the item that is in error is identified" BUT NOT "the error is described to the user in text"!

**So exactly the opposite of the previous example.** !!!

Will this FAIL? because not described? Only 50% of the SC is done properly?

<img width="518" alt="Screen Shot 2021-05-17 at 11 01 35 AM" src="https://user-images.githubusercontent.com/6384002/118462759-5130cb00-b6ff-11eb-8d5d-65077b9ab94c.png">

**Comment by patrickhlauke** [comment]
> QUESTION: This IS "the error is described to the user in text" BUT is "the item that is in error is identified to the user in text"?

arguably, the item that is in error is identified in text by the fact that the error message appears next/underneath it, thus identifying it as having an error (don't think the wording means that the text must explicitly use words to say "this field name [NAME] is in error"

> QUESTION: This IS "the item that is in error is identified" BUT NOT "the error is described to the user in text"!

yeah this butts up against what/how specific the "description" needs to be. I think I tried to make a start tackling this prickly handwavy topic (and how it overlaps with 3.3.3) in https://github.com/w3c/wcag/pull/1651

**Comment by jake-abma** [comment]
> arguably, the item that is in error is identified in text by the fact that the error message appears next/underneath it, thus identifying it as having an error (don't think the wording means that the text must explicitly use words to say "this field name [NAME] is in error"


I kind of agree, but the "AND" word suggests / means it's 2 things to do, it does not says:

> the item that is in error is identified by describing to the user in text.

We often do a lot of hard work with SC wording to make sure the 'and' is really 2 things, but not now... when we discuss this in training people are confused... even I am, although we all get the intent...

**Comment by patrickhlauke** [comment]
It says it's two things, because they are. They *can* be satisfied separately. If you added just the extra word "Error" next to the field (to identify it) and then somewhere else on the form (e.g. in a block right at the start of the form) had text that said "The [BLAH] field has been left empty" to describe the error, you've passed the SC as well. i.e. the indication and description can, but don't have to, be the same text.

**Comment by jake-abma** [comment]
sound logical, and agree, but isn't "required" for a field not the same, but in other words as "Ingredients must not be empty"?

But in this case, 'because you mention it extra / again / in other words' it counts as saying "describe the error in text" ... ?

**Comment by patrickhlauke** [comment]
it's also the proximity here that counts. even if the error message was just "this field must not be empty", the fact that it's near it "indicates" that it refers to that.

**Comment by patrickhlauke** [comment]
> sound logical, and agree, but isn't "required" for a field not the same

but that word "Required" is already there even when not in error. so you're not really indicating anything (unless you want to pass it for indicating, and then failing it under use of color for yes indicating the error, but just relying on color for that)

**Comment by jake-abma** [comment]
I know, but that is not what I mean, I meant that it basically says the same, but in other words... (not talking about the color) and it just repeats... it does not provide extra description

**Comment by patrickhlauke** [comment]
fourth bullet point of my proposed PR https://github.com/w3c/wcag/pull/1651

> add allowance for situations where an error indication is already self-explanatory/obvious from context (e.g. a form where fields have already explicitly been identified as mandatory/required - not necessary for compliance to now ALSO explicitly say "as we already told you, this field is mandatory")
